### PR TITLE
ShareModal: Responsive tabs and Public Dashboards modal fixed

### LINKED ISSUE #64318: Task: Public dashboard modal not working for mobile
The modal which has the public dashboard tab is not responsive, so it's impossible to click on it.
It's important to notice the share button (which opens the modal) is not shown in mobile. 
However, it's possible to access the modal through the audit table

### PR DESCRIPTION
**Pull request by juanicabanas**
**What is this feature?**

- It makes share modal tabs responsive, so they can be horizontally scrolled.
- It fixes public dashboard modal to be responsive as well.
- If refactors TabsBar component to use useStyles2 and not to pass boolean so it's memoized
- It changes flex-grow to flex in order to guarantee the sibling isn't affected

**Which issue(s) does this PR fix?**:

Fixes [#64318](https://github.com/grafana/grafana/issues/64318)

**Special notes for your reviewer:**

**Before**

https://user-images.githubusercontent.com/11707172/232875226-468d6725-94a6-4798-aef6-123c9b4072c3.mov

**Now**

https://user-images.githubusercontent.com/11707172/232874733-22d54c51-ad67-4615-8422-ebb5ce9d7eab.mov

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

**Comment by torkelo** [comment]
Does not seem like the ideal way to make tabs responsive, another common way is to turn the tabs into a menu or stack them (which is not great if they are many). But I think it's hard to make this responsive behavior fully generic (apply to all situations where we have a tabs). Would love some thoughts here from frontend platform & design system squad what our options are here and how easy / hard they are to generalize. 

Maybe reuse / copy some of the logic we have for ToolbarButtonRow (it collapses into a menu progressively as each button don't fit)  is also an option, but not sure that is great either this is a bit different (as you have a "current state" = current tab, and you don't have that in the ToolbarButtonRow)

**Comment by juanicabanas** [comment]
> Does not seem like the ideal way to make tabs responsive, another common way is to turn the tabs into a menu or stack them (which is not great if they are many). But I think it's hard to make this responsive behavior fully generic (apply to all situations where we have a tabs). Would love some thoughts here from frontend platform & design system squad what our options are here and how easy / hard they are to generalize.
> 
> Maybe reuse / copy some of the logic we have for ToolbarButtonRow (it collapses into a menu progressively as each button don't fit) is also an option, but not sure that is great either this is a bit different (as you have a "current state" = current tab, and you don't have that in the ToolbarButtonRow)

Got it. Any thoughts? @grafana/grafana-frontend-platform @grafana/design-system

**Comment by juanicabanas** [comment]
Would be good to fix it as soon as possible, since I'm working in another feature for G10 which has the same problem 
image

**Comment by staton-hyse11** [comment]
I'm wondering if it would be possible to transition the tabs to our 'select' once we run out of space - or something similar. Maybe this becomes the responsive behavior of the tabs to transition into a dropdown-like element at a particular screen size. This would maintain the current location as a selected state in the dropdown. 

With the 'ToolbarButtonRow', we would need to solve for the strange stacking and how the 'active' state would work.

image

**Comment by staton-hyse11** [comment]
Here is a found example of my initial idea:
image

**Comment by juanicabanas** [comment]
> Here is a found example of my initial idea: finished-working-tab-behavior [ finished-working-tab-behavior ](https://user-images.githubusercontent.com/101748563/233460499-fdcbb5c9-001f-498b-bd16-7fa19aecee6b.gif) [ ](https://user-images.githubusercontent.com/101748563/233460499-fdcbb5c9-001f-498b-bd16-7fa19aecee6b.gif)

This is a good idea but we should make sure every component renders good in this way. I just made a quick POC and, for example, in my case I would need to modify my component to put this new select below and not next to the "Share" title in the modal header. 

On the other hand, a `Tab` can display an icon, a counter or a suffix (or all of them) next to the label. Will it be good to render in a `select` item? 
And I just checked and there's an example where `TabsBar` component not only receives `Tabs` as children, but also another component, such as an `IconButton`.

With all of this, I think it's not possible to have a solution previous G10. At least I'm working in other stuffs and I won't be able to work on this.

**Comment by torkelo** [comment]
Yes , using a select is a common pattern. One we used in the old nav page tabs

**Comment by natellium** [comment]
Chiming in here: given as @juanicabanas said that the select approach is not doable for G10 and the current implementation is unusable, couldn't we go for the "scrollable" option as an interim solution?

**Comment by staton-hyse11** [comment]
> Chiming in here: given as @juanicabanas said that the select approach is not doable for G10 and the current implementation is unusable, couldn't we go for the "scrollable" option as an interim solution?

That is an approach but it would need a solution for when the tabs are not partially cut-off. 

This doesn't provide any indication there are other tabs that can be scrolled into view
image

The addition of chevrons could help and could also have a "click to see next" behavior

I would also recommend that this future work gets flagged so it's not lost after G10

**Review by axelavargas — APPROVED** [review]
Hey everyone 😄 , although this PR may not fulfill all user expectations, I think we can consider it as an intermediate step for G10, particularly because the current behavior doesn't allow users to access any options. 

By approving this PR, we can at least enable users to access options on mobile devices. We could create a new issue to further address this problem and tackle it after G10.

File: packages/grafana-ui/src/components/Tabs/TabsBar.tsx [hunk 1]
```diff
@@ -13,32 +13,30 @@ export interface Props {
...
-export const TabsBar = React.forwardRef<HTMLDivElement, Props>(({ children, className, hideBorder }, ref) => {
-  const theme = useTheme2();
-  const tabsStyles = getTabsBarStyles(theme, hideBorder);
+export const TabsBar = React.forwardRef<HTMLDivElement, Props>(({ children, className, hideBorder = false }, ref) => {
+  const styles = useStyles2(getStyles);
 
   return (
-    <div className={cx(tabsStyles.tabsWrapper, className)} ref={ref}>
-      <div className={tabsStyles.tabs} role="tablist">
+    <div className={cx({ [styles.tabsBorderBottom]: !hideBorder }, styles.tabsScroll, className)} ref={ref}>
```
Comment (torkelo) [inline packages/grafana-ui/src/components/Tabs/TabsBar.tsx:20]:
```suggestion
    <div className={cx(styles.tabsWrapper, hideBorder && styles.noBorder, className)} ref={ref}>
```

tabsWrapper has the scroll

Comment (juanicabanas) [reply packages/grafana-ui/src/components/Tabs/TabsBar.tsx:20]: you mean to have a tabsWrapper which is
tabsWrapper: css `
border-bottom: 1px solid ${theme.colors.border.weak};
  overflow-x: auto;
`

and then another property noBorder:
noBorder: css `
border-bottom: 0;
`
?

Comment (torkelo) [reply packages/grafana-ui/src/components/Tabs/TabsBar.tsx:20]: yes

Comment (juanicabanas) [reply packages/grafana-ui/src/components/Tabs/TabsBar.tsx]: done. I need to check a test that is failing though

**Review by torkelo — APPROVED** [review]

**Pull request merged**